src/unix/fs.c

```diff
@@ -231,7 +231,68 @@ static ssize_t uv__fs_futime(uv_fs_t* req) {
 
 
 static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
+#if defined(__MVS__)
+  /* There is no mkdtemp. So instead use mktemp to generate a
+     temporary file. Then delete that file and use the name
+     to create a temporary directory.
+  */
+  char *path = (char*) req->path;
+  static const char* tempchars =
+    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  static const size_t num_chars = 62;
+  static const size_t num_x = 6;
+  char *ep, *cp;
+  unsigned int tries, i;
+  size_t len;
+  uint64_t v;
+  FILE *finput;
+  int retval;
+  int saved_errno;
+
+  len = strlen(path);
+  ep = path + len;
+  if (len < num_x || strncmp(ep - num_x, "XXXXXX", num_x)) {
+    errno = EINVAL;
+    return -1;
+  }
+
+  finput = fopen("/dev/urandom", "r");
```

Comment:
You sure you want to use `fopen()` and not `open(2)`/`read(2)`? I generally avoid stdio if I'm not doing formatted I/O, but maybe that is just me. Ben? Saghul?  The BSD code this is derived from, http://web.mit.edu/freebsd/head/lib/libc/stdio/mktemp.c ?,  uses [arc4random()](https://www.freebsd.org/cgi/man.cgi?query=arc4random&sektion=3), which only seeds once from urandom, but I guess that's just an optimization that could be applied later. z doesn't have an arc4random() or [getrandom()](http://man7.org/linux/man-pages/man2/getrandom.2.html)? I also wonder if this would be better moved to an os-390-mkdtemp.c, rather than inline ifdefs. Btw, I'm just kibitzing, Ben and Saghul have a better handle on uv, its just that I had some comments on the original mkdtemp code merge which introduced a license violation, so I noticed when yet another mkdtemp implementation showed up.